components/layout_2020/geom.rs

```diff
@@ -754,7 +754,18 @@ impl LogicalVec2<Size<LengthPercentage>> {
         LogicalVec2 {
             inline: self
                 .inline
-                .maybe_map(|v| v.maybe_to_used_value(basis.inline))
+                .maybe_map(|v| {
+                    // Resolves child box inside parent with percentage width css style has larger width and will be
+                    // splited to multiply lines. It is a solution to avoid the problem of round and truncated method
+                    // on Au and CSSPercentage.
+                    if v.to_percentage().is_some() && basis.inline.is_some() {
+                        Some(Au::from_f32_px(
+                            v.to_pixel_length(basis.inline?).px().floor(),
+                        ))
```

Comment:
```suggestion
                    if v.to_percentage().is_some() {
                        basis.inline.map(|inline| Au::from_f32_px(v.to_pixel_length(inline).px().floor()))
```

I'm not familiar with this code, so I can't comment on the correctness, but I believe we can simplify your code by using map

Reply:
I'm not familiar with rust, do you mean the following usage?

```rust
v.to_percentage().map_or_else(
    || v.maybe_to_used_value(basis.inline),
    |_| {
        Some(Au::from_f32_px(
            v.to_pixel_length(basis.inline?).px().floor(),
        ))
    },
)
```